apps/push-service/src/push/router/stream.ts

```diff
@@ -255,94 +266,86 @@ export const createStreamRouter = (
-                        response = await axios.post(endpointWebsocket, next);
-                        callResponseTime = new Date().getTime() - beforeWebhook;
-                      }
-                    } catch (err) {
-                      response.statusText = err.message;
-                      response.status = 400;
-                      response.headers = { date: new Date() };
-                      logger.info(`Failed sending request from status.`);
-                      logger.info(`Error: ${JSON.stringify(err.message, getCircularReplacer())}`);
-                      callResponseTime = new Date().getTime() - beforeWebhook;
-                    } finally {
-                      eventService.send(
-                        webhookTriggered(
-                          user,
-                          tenantId,
-                          webhook.url,
-                          webhook.targetId,
-                          webhook.eventTypes,
-                          webhook.name,
-                          response?.statusText,
-                          response?.status,
-                          response?.headers?.date,
-                          callResponseTime
-                        )
-                      );
-                    }
-                  }
-                });
-              });
-          } catch (e) {
-            console.error('Error: ' + JSON.stringify(e));
+      const response = await configurationService.getConfiguration<Record<string, Webhook>, Record<string, Webhook>>(
+        serviceId,
+        token,
+        tenantId
+      );
+
+      const webhooks = response?.webhooks;
+
+      Object.keys(webhooks).map(async (key) => {
+        const eventMatches = [];
+        const webhook = webhooks[key] as Webhook;
+        const eventTypes = webhook.eventTypes;
+
+        eventTypes.map((et) => {
+          const nextPayload = next.payload as unknown as NextPayload;
+          if (`${next.namespace}:${next.name}` === et.id && et.id !== 'push-service:webhook-triggered') {
+            if (
+              (nextPayload?.application?.appKey && nextPayload?.application?.appKey === webhook.targetId) ||
+              (nextPayload?.application?.id && nextPayload?.application?.id === webhook.targetId)
+            ) {
+              next.payload;
+              eventMatches.push(et.id);
+            }
           }
         });
+        const endpointWebsocket = webhooks[key].url;
+
+        if (eventMatches.length > 0) {
+          let response: StatusResponse = {};
+          let callResponseTime = 0;
+          const beforeWebhook = new Date().getTime();
+          try {
+            if (isValidUrl(endpointWebsocket)) {
+              response = await axios.post(endpointWebsocket, next);
+              callResponseTime = new Date().getTime() - beforeWebhook;
+            }
+          } catch (err) {
+            response.statusText = err.message;
+            response.status = 400;
```

Comment:
Ideally, the error response status shall be based on the error type. We can think of how to improve this laster.